### PR TITLE
Circuit breaker improvement

### PR DESCRIPTION
**Pull request by tbuchaillot**
Tyk's Circuit Breaker only works for 500 errors. This is pretty limiting. e.g. the breaker will not trip for failed upstreams.

This PR fixes that in order to break the circuit for any 5XX errors or if RoundTrip returns an error.

### CODE REVIEW

File: gateway/reverse_proxy_test.go [hunk 1]
```diff
@@ -347,6 +348,38 @@ func TestWrappedServeHTTP(t *testing.T) {
...
 }
 
+func TestCircuitBreaker5xxs(t *testing.T) {
+	ts := StartTest()
+	defer ts.Close()
+
+	t.Run("Extended Paths", func(t *testing.T) {
+		BuildAndLoadAPI(func(spec *APISpec) {
+			UpdateAPIVersion(spec, "v1", func(v *apidef.VersionInfo) {
+				json.Unmarshal([]byte(`[
```
Comment (buger) [inline gateway/reverse_proxy_test.go]: Feels like formatting error.

**Comment by buger** [comment]
Apart from small formatting bug, it works

File: gateway/reverse_proxy_test.go (hunk 1 above)
Comment (tbuchaillot) [reply gateway/reverse_proxy_test.go]: fixed

**Pull request merged**